namesys/namesys.go

```diff
@@ -87,9 +89,44 @@ func (ns *mpns) resolveOnce(ctx context.Context, name string) (path.Path, error)
 
 // Publish implements Publisher
 func (ns *mpns) Publish(ctx context.Context, name ci.PrivKey, value path.Path) error {
-	return ns.publishers["/ipns/"].Publish(ctx, name, value)
+	err := ns.publishers["/ipns/"].Publish(ctx, name, value)
+	if err != nil {
+		return err
+	}
+	ns.addToDHTCache(name, value, time.Now().Add(time.Hour*24))
+	return nil
+}
+
+func (ns *mpns) PublishWithEOL(ctx context.Context, name ci.PrivKey, value path.Path, eol time.Time) error {
+	err := ns.publishers["/ipns/"].PublishWithEOL(ctx, name, value, eol)
+	if err != nil {
+		return err
+	}
+	ns.addToDHTCache(name, value, eol)
+	return nil
 }
 
-func (ns *mpns) PublishWithEOL(ctx context.Context, name ci.PrivKey, val path.Path, eol time.Time) error {
-	return ns.publishers["/ipns/"].PublishWithEOL(ctx, name, val, eol)
+func (ns *mpns) addToDHTCache(key ci.PrivKey, value path.Path, eol time.Time) {
+	var err error
+	value, err = path.ParsePath(value.String())
```

Comment:
Yes as the publisher can publish hash only but resolver will sanitize this value to be `/ipfs/CID`.